ansible/roles/grafana/tasks/post_config.yml

```diff
@@ -4,7 +4,7 @@
     url: "{{ internal_protocol }}://{{ kolla_internal_vip_address }}:{{ grafana_server_port }}/login"
     status_code: 200
   register: result
-  until: result | failed or result.status == 200
+  until: result.status == 200
```

Comment:
Hmm, I guess we should retry on connection failure. I thought you had an error where status was not set previously though?

Reply:
I did indeed :

```
TASK [grafana : Wait for grafana application ready] ***************** fatal: [some_remote_node]: FAILED! => {"failed": true, "msg": "The conditional check 'result.status == 200' failed. The error was: error while evaluating conditional (result.status == 200): 'dict object' has no attribute 'status'"}
```
I've written a minimal playbook and I can't reproduce it now (same version of ansible).

Yeah, there are cases when retrying is a waste of time, and cases when it's worth doing.

Reply:
As I remember, the specific case you had was an unrouteable IP - accessing the public VIP from the director at VG?

Reply:
That's correct. I've tried this playbook (with an unroutable IP). 

```
---
- hosts: monitoring
  remote_user: stack
  tasks:
    - name: Wait for grafana application ready
      uri:
        url: "http://11.2.2.3:3000/login"
        status_code: 200
      register: result
      until: result.status == 200
      retries: 10
      delay: 2
      run_once: true
```

Reply:
How about using `result.get('status') == 200`, to be sure?